src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -125,13 +125,16 @@ case class CsvRelation protected[spark] (
           case aiob: ArrayIndexOutOfBoundsException if permissive =>
             (index until schemaFields.length).foreach(ind => rowArray(ind) = null)
             Some(Row.fromSeq(rowArray))
-          case _: java.lang.NumberFormatException |
-               _: IllegalArgumentException if dropMalformed =>
-            logger.warn("Number format exception. " +
+          case e: java.lang.NumberFormatException if dropMalformed =>
+            logger.warn("Number format exception  (" + e + ")." +
+              s"Dropping malformed line: ${tokens.mkString(delimiter.toString)}")
+            None
+          case e: IllegalArgumentException if dropMalformed =>
+            logger.warn("IllegalArgument exception  (" + e + ")." +
               s"Dropping malformed line: ${tokens.mkString(delimiter.toString)}")
             None
           case pe: java.text.ParseException if dropMalformed =>
-            logger.warn("Parse exception. " +
+            logger.warn("Parse exception. (" + pe + ")." +
```

Comment:
Personal opinion: since this is a warning message, shouldn't we maybe just print out the message only by `pe.getMessage()`?